tests/statistics/stats_utils.py

```diff
@@ -0,0 +1,105 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) Pootle contributors.
+#
+# This file is a part of the Pootle project. It is distributed under the GPL3
+# or later license. See the LICENSE file for a copy of the license and the
+# AUTHORS file for copyright and authorship information.
+
+from collections import OrderedDict
+from datetime import datetime
+
+import pytest
+
+from pytest_pootle.factories import UserFactory
+
+from django.contrib.auth import get_user_model
+from django.db.models import Q
+
+from pootle.core.delegate import contributors
+from pootle_statistics.models import Submission, SubmissionTypes
+from pootle_statistics.utils import Contributors
+from pootle_store.models import Store
+
+
+User = get_user_model()
+
+
+def _contributors_list(contribs,
+                       projects=None, languages=None,
+                       since=None, until=None):
+    subs = []
+    for user in contribs.user_qs.distinct():
+        q = Q(submitter=user)
+        if projects:
+            q = q & Q(
+                translation_project__project__code__in=projects)
+        if languages:
+            q = q & Q(
+                translation_project__language__code__in=languages)
+        if since or until:
+            _q = Q()
```

Comment:
im not 100% sure if it necessary (i think it is - it seemed to be collapsing the query before - not sure) but it structures the query differently